dev-libs/libprelude/libprelude-4.0.0.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6}} )
+
+USE_RUBY="ruby21 ruby22 ruby23 ruby24"
+
+inherit autotools python-r1 ruby-single
+
+DESCRIPTION="Prelude-SIEM Framework Library"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc lua python ruby perl"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="net-libs/gnutls:=
+	perl? ( dev-lang/perl:= virtual/perl-ExtUtils-MakeMaker )
+	lua? ( dev-lang/lua:* )
+	ruby? ( ${RUBY_DEPS} )
+	python? ( ${PYTHON_DEPS} )"
+
+DEPEND="${RDEPEND}
+	doc? ( dev-util/gtk-doc )
+	sys-devel/flex
+	virtual/yacc
+	>=dev-lang/swig-3.0.7
+	virtual/pkgconfig"
+
+src_prepare() {
+	default
+
+	# Avoid null runpaths in Perl bindings.
+	sed -e 's/ LD_RUN_PATH=""//' -i "${S}/bindings/Makefile.am" || die "sed failed"
+
+	eautoreconf
+}
+
+src_configure() {
+	local python2_configure=--without-python2
+	local python3_configure=--without-python3
+
+	chk_python() {
+		if [[ ${EPYTHON} == python2* ]]; then
+			python2_configure=--with-python2
+		elif [[ ${EPYTHON} == python3* ]]; then
+			python3_configure=--with-python3
+		fi
+	}
+
+	if use python; then
+		python_foreach_impl chk_python
```

Comment:
There was a long discussion whan I pushed libprelude 3.0 and 3.1 about this and this code is the conclusion.

Can you tell me what you have in mind more precisely ?

about build accidentally, remember that I put  this at the top of the ebuild:
REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"

Reply:
Please look through the docs @ https://wiki.gentoo.org/wiki/Project:Python. They explain how to use python-r1 eclass in some detail.

Reply:
Thats what I did before but for what I need, I didn't find better. Please help me on this.

The thing is if the system have python3.4 installed (for example) but you don't want Prelude to be compiled with it, you have to specifie --without-python3 else, with no configure flag, the configure will  detect python3 support, so t will build it.

So for you, what should I have in this file ?

Thansk for you help

Reply:
The first question is, whether you really want to support building for multiple implementations simultaneously? If not, this is as easy as inheriting python-single-r1, letting `python-single-r1_pkg_setup` take care of selecting one implementation and doing `if python_is_python3` to set either python2 or python3 depending on the one selected. You will also have to set `DISTUTILS_SINGLE_IMPL` in the ebuild using distutils.

If you insist on doing multi-impl, then you will need to configure the basic version without python at all, then add `python_foreach_impl python_configure` and so on, and make those functions configure, build and install additional subvariants for each Python implementation. Depending on the package internals, this might be as simple as calling `emake -C python` or something like that to build only Python subdirectory, or as complex as having to play with automake internals such as `VPATH`.

Reply:
everywhere, we provide all prelude python implementation that the user ask for so I want to provide this. And before on gentoo, we provide it so I don't want to make regression.

So, in your second paragraph, finally, you just want that I rename my function chk_function to python_configure and put it out of src_configure() ?
To enable python2 or python3 support, I just have to add --with-python2 or --with-python3 to my configure options. This is not a simple python module with setup.py.

Reply:
No. You pretend to provide all python implementations while actually max two of them are installed. You provide extra fake flags that may or may not work, and two prelude components with the same flags may end up using two totally random python implementations that don't even match.

No. I want you to build the Python support *separately* for *every* Python implementation. By adding `--with-python2`/`3` you only make the build system attempt to build *one* random implementation.

```
# tar -tf /srv/nfs/common/packages/athlon64/dev-libs/libprelude/libprelude-4.0.0-1.xpak | grep python
./usr/lib64/python3.5/
./usr/lib64/python3.5/site-packages/
./usr/lib64/python3.5/site-packages/_prelude.cpython-35m-x86_64-linux-gnu.so
./usr/lib64/python3.5/site-packages/prelude-4.0.0-py3.5.egg-info
./usr/lib64/python3.5/site-packages/prelude.py
./usr/lib64/python2.7/
./usr/lib64/python2.7/site-packages/
./usr/lib64/python2.7/site-packages/_prelude.so
./usr/lib64/python2.7/site-packages/prelude-4.0.0-py2.7.egg-info
./usr/lib64/python2.7/site-packages/prelude.py
```

yet I had 2.7+3.4+3.5+3.6 enabled. If you want to argue about something, then please at least bother testing it.

Reply:
Hum, I'm really sorry, It was working before when I did previous ebuild but it obviously didn't work anymore, strange. That's why we didn't understand each other ....

OK, so my goal is to have prelude for python 3.4, 3.5 and 3.6 installed in the right folder if you provide python3 useflag and have python 3.4, 3.5. 3.6 installed.

I will check this and get back to you ...

Can you tell me your default python so I can set my system in the same way than yours ?

Reply:
There is no such thing as 'default python' anymore. My preference list is:

    python3.5
    python3.4

Reply:
I'd like to note that we asked this in #gentoo-python about a year ago and there were no suggestions back then:

```
2016-07-29 22:45:21     gokturk is there a smart way to obtain python implementations in a system other than running python_foreach_impl? _python_obtain_impls is marked as internal.
2016-07-29 22:46:48     gokturk needed for --with-python2/--with-python3 type configure
2016-07-29 22:47:35     gokturk nothing uses python2/python3 USE flags other than portage, so I'm guessing there's a preferred way to do this.
2016-07-30 05:53:55     @mgorny gokturk, nah, that is rare enough not to have a dedicated function
2016-07-30 05:54:08     @mgorny Otherwise, people would abuse it to do wrong things
2016-07-30 06:03:44     ToToL   gokturk: I also search for it but this is the only way I found to do it and not seems too much hacky
2016-07-30 06:03:59     ToToL   thanks for review, I will do it
```

Reply:
You *asked* how to get impls. You didn't mention how you were going to use it. Yet I find my 'people would abuse it to do wrong things' surprisingly prophetic because here we are.

Reply:
Here is a new proposition. In my system, I get all prelude lib for all present python version.

What do you think now ?